lib/product.rb

```diff
@@ -0,0 +1,26 @@
+class Product
+  attr_reader :category, :name , :unit_price, :quantity, :hoarded
+
+  def initialize(category, name, unit_price, quantity)
+    @category = category
+    @name = name
+    @unit_price = unit_price
+    @quantity = quantity.to_i
+    @hoarded = hoarded
```

Comment:
You have some recursion happening here. You're setting it equal to the method that calls this variable. Instead, set it to the actual default value - false.